### PR TITLE
Implemented Room reports, suspensions, and terminations.

### PR DESCRIPTION
**Pull request by bubby932**
* Bumped a package using Dependabot.
* Added the Room Report endpoint.
* Added the moderation-suspend room endpoint.
* Added the moderation-terminate room endpoint.
* Fixed an issue where developers didn't have proper permissions in the search menu.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "express-fileupload": "^1.2.1",
     "express-rate-limit": "^6.7.0",
     "express-slow-down": "^1.5.0",
-    "firebase": "^9.14.0",
+    "firebase": "^9.15.0",
```
Comment (bubby932) [inline package.json:23]: @depentabot Bumped.

File: routers/rooms.js [hunk 2]
```diff
@@ -36,7 +36,9 @@ const AuditEventType = {
     /** MAJOR - Logged when a developer suspends this room, locking everyone out until moderation review concludes. */
     RoomSuspendedByDeveloper: "room_suspended",
     /** MAJOR - Logged when a developer terminates this room, permanently locking it from all users. */
-    RoomTerminatedByDeveloper: "room_terminated"
+    RoomTerminatedByDeveloper: "room_terminated",
```
Comment (bubby932) [inline routers/rooms.js:39]: More audit events.

File: routers/rooms.js [hunk 3]
```diff
@@ -163,7 +165,7 @@ router.get("/search", authenticateToken_optional, async (req, res) => {
         if(typeof req.user != 'undefined') {
             const role = Object.keys(userPermissions).includes(req.user.id) ? userPermissions[req.user.id] : "everyone";
             const permissions = rolePermissions[role];
-            return permissions.viewAndJoin;
+            return permissions.viewAndJoin || req.user.developer;
```
Comment (bubby932) [inline routers/rooms.js:168]: Fixed an issue where developers didn't have enough permissions.

File: routers/rooms.js [hunk 4]
```diff
@@ -658,6 +660,7 @@ router.post("/room/:id/moderation-terminate", authenticateDeveloperToken, async
     try {
         const { id: room_id } = req.params;
         const { note } = req.body;
+        const { permanent } = req.query;
```
Comment (bubby932) [inline routers/rooms.js:663]: Added the `permanent` option to room termination, for legal purposes.

File: routers/rooms.js [hunk 5]
```diff
@@ -890,6 +916,134 @@ router.get('/room/:id/subrooms/:subroom_id/versions', authenticateToken, canView
     }
 });
 
+const ReportRateLimit = rateLimit({
```
Comment (bubby932) [inline routers/rooms.js:919]: Added room reports and a rate limit to go along with it.

**Review by bubby932 — COMMENTED** [review]
Explained changes.

**Review by bubby932 — COMMENTED** [review]
updated review

File: routers/rooms.js [hunk 6]
```diff
@@ -890,6 +916,134 @@ router.get('/room/:id/subrooms/:subroom_id/versions', authenticateToken, canView
...
+            auditLog(
+                `
+                !! EMERGENCY !!
+                <@&812976292634427394>
+                A player (ID ${req.user.id}) has submitted a report against room '${data.name}' (ID ${data._id})!
+                Reason: 
+                \`${reason}\`
+                HOWEVER
+                The user also indicated that this room may pose an ***IMMEDIATE THREAT TO HUMAN LIFE!***
+                It is absolutely paramount that this room is immediately investigated! Serious IRL consequences may result if it is not!
```
Comment (Bobrobot1) [inline routers/rooms.js:1019]: Don't use acronyms like IRL

**Review by Bobrobot1 — CHANGES_REQUESTED** [review]
See note

File: routers/rooms.js (hunk 6 above)
Comment (bubby932) [reply routers/rooms.js:1019]: this only gets shown to developers though

Comment (Bobrobot1) [reply routers/rooms.js:1019]: Still, it's important to spell things out when dealing with potentially illegal content.

Comment (bubby932) [reply routers/rooms.js]: see update

**Review by Bobrobot1 — APPROVED** [review]

**Pull request merged**